src/analytics/AsyncStoragePersistor.ts

```diff
@@ -0,0 +1,34 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'AsyncStoragePersistor'
+/**
+ * Persistor implementation using AsyncStorage
+ * From https://github.com/segmentio/analytics-react-native/blob/master/packages/sovran/src/persistor/async-storage-persistor.ts
+ *
+ * This is almost exactly the same as the default persistor that segment uses however our build was not liking the persistor linked above due to its
+ * use of require('@react-native-async-storage/async-storage'). The require was resolving to { default: AsynctStorage } instead of just AsyncStorage
+ * so calls to AsyncStorage.getItem and AsyncStorage.setItem were not happening because they were undefined. This is a workaround to that issue.
```

Comment:
this looks fine, but would a patch work for this instead? Seems like it might be a reasonable use case for one